src/data/nat/digits.lean

```diff
@@ -236,10 +238,71 @@ begin
   { simp [of_digits, list.sum_cons, ih], }
 end
 
-/-! ### Inequalities
+/-!
+### Properties
+
+This section contains various lemmas of properties relating to `digits` and `of_digits`.
+-/
+
+lemma digits_eq_nil_iff_eq_zero (b n : ℕ) : digits b n = [] ↔ n = 0 :=
+begin
+  split,
+  { intro h,
+    have : of_digits b (digits b n) = of_digits b [], by rw h,
+    convert this,
+    rw of_digits_digits },
+  { rintro rfl,
+    simp }
+end
+
+lemma digits_ne_nil_iff_ne_zero (b n : ℕ) : digits b n ≠ [] ↔ n ≠ 0 :=
+⟨λ h, h ∘ (digits_eq_nil_iff_eq_zero _ _).mpr,
+ λ h, h ∘ (digits_eq_nil_iff_eq_zero _ _).mp⟩
+
+private lemma digits_last_aux1 {α : Type*} {L M : list α} (h : L = M) (p q) : L.last p = M.last q :=
+by simp [h]
 
-This section contains various lemmas with inequalities relating to `digits` and `of_digits`.
- -/
+private lemma digits_last_aux2 (b n : ℕ) (h : 2 ≤ b) (w : 0 < n) :
+  digits b n = ((n % b) :: digits b (n / b)) :=
+begin
+  rcases b with _|_|b,
+  { finish },
+  { norm_num at h },
+  rcases n with _|n,
+  { norm_num at w },
+  simp,
+end
+
+lemma digits_last (b : ℕ) (h : 2 ≤ b) (m : ℕ) (hm : 0 < m) (p q) :
+  (digits b m).last p = (digits b (m/b)).last q :=
+begin
+  rw digits_last_aux1 (digits_last_aux2 b m h hm),
+  { rw list.last_cons, },
+  exact list.cons_ne_nil _ _,
+end
```

Comment:
```suggestion
by { simp only [digits_last_aux2 b m h hm], rw list.last_cons }
```
You can then also remove `digits_last_aux1` (simp is very good at dealing this propositional and subsingleton arguments for functions).